src/psbt.js

```diff
@@ -231,6 +234,135 @@ export function psbtOutputFormatter(output) {
   }
 }
 
+/**
+ * Translates a PSBT into inputs/outputs consumable by non-PSBT devices in the
+ * `unchained-wallets` library.
+ *
+ * FIXME - Have only confirmed this is working for P2SH addresses on Ledger on regtest
+ *
+ * @param {module:networks.NETWORKS} network - bitcoin network
+ * @param {module:multisig.MULTISIG_ADDRESS_TYPES} addressType - address type
+ * @param {String} psbt - PSBT as a base64 or hex string
+ * @param {Object} keyDetails - Object containing signing key details (Fingerprint + bip32 prefix)
```

Comment:
Are key details the same as "ExtendedPublicKeyDerivation" (or maybe `KeyDerivation` to be more generic)? 

In past discussions (and in the bitcoinjs library from which this is based) we've defined this as something like:

```
interface PublicKeyDerivation {
  rootFingerprint: Buffer | string;
  pubkey: Buffer | string; 
  path: string;
}
```

the xpub derivation being the same except `xpub` instead of `pubkey`. Maybe the base "class" could be:

```
interface KeyDerivation {
  rootFingerprint: Buffer | string // alternatively could be xfp?
  path: string
}
```

Reply:
No, not really. It happens to be compatible with the `KeyDerivation` you defined, but in this case it's not really what we normally mean. e.g. the path we're providing is not the path to the public key but rather the root path.

Here's the problem:

In the general case we should only need to provide the `rootFingerprint` || `masterFingerprint` || `xfp` to know which device is trying to sign, HOWEVER -- in the case where you put multiple keys from the same device in the same multisig ... aka lots of test cases everywhere ... we need to be able to differentiate who is tryting to sign and this is a way to do that.

Reply:
I'm not totally clear on what `rootPath` is? The first thing that comes to mind is the path of the root key, but that would just be `m` of course and irrelevant/redundant. Next is the path from which the xpub is derived. But isn't that just the path that's in the `ExtendedPublicKeyDerivation`?

```
interface ExtendedPublicKeyDerivation {
    rootFingerprint: Buffer | string;
    xpub: string;
    path: string;
}
```
In the [descriptor spec](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#key-origin-identification), this grouping is called the `KeyOrigin`. To add to the confusion, in the bip174/bitcoinjs library this is `Bip32Derivation` ([source](https://github.com/bitcoinjs/bip174/blob/master/ts_src/lib/interfaces.ts#L85))

```
export interface Bip32Derivation {
  masterFingerprint: Buffer;
  pubkey: Buffer;
  path: string;
}
```

Maybe I'm just misunderstanding something? Or another way to put it is you're just trying to make a distinction between a path to a public key that's used to generate an address and the path used to _find_ the origin of the xpub? In which case, I think we're saying the same thing. Basically there's a derivation for a pubkey which would have a path to get that, and a derivation for an xpub which would be used to get that extended pubkey. 

My main objection is just adding more new vocabulary, especially in proptypes where that's going to be important, especially if it's not in what's being used elsewhere. This has already been a really confusing aspect of working with this library in the past. 

The name of the variable isn't ultimately too important b/c it can be changed, though my preference is to stick to some of what we've already talked about and have been using, which is above. For the properties, I think it should just be `path` or `bip32Path`. `root` feels very confusing to me and isn't used in other places i've seen, and in the context of an xpub derivation, the path could only really refer to the "origin" of that xpub (as it's used in descriptors).

Reply:
Sure, can rename it - hopefully makes it clearer.